apps/simple-trading-app/src/app/components/simple-market-list/simple-market-list.tsx

```diff
@@ -1,28 +1,49 @@
-import React, { useCallback, useMemo } from 'react';
+import React, { useCallback, useEffect, useMemo, useRef } from 'react';
 import { useNavigate } from 'react-router-dom';
+import { subDays } from 'date-fns';
 import { useDataProvider } from '@vegaprotocol/react-helpers';
 import { t } from '@vegaprotocol/react-helpers';
 import { AsyncRenderer, Lozenge, Splash } from '@vegaprotocol/ui-toolkit';
 import { Button } from '@vegaprotocol/ui-toolkit';
+import type { MarketState } from '@vegaprotocol/types';
 import SimpleMarketPercentChange from './simple-market-percent-change';
 import SimpleMarketExpires from './simple-market-expires';
 import DataProvider from './data-provider';
 import { MARKET_STATUS } from './constants';
 
 const SimpleMarketList = () => {
   const navigate = useNavigate();
+  const statusesRef = useRef<Record<string, MarketState | ''>>({});
   const variables = useMemo(
     () => ({
-      CandleInterval: 'I1H',
-      CandleSince: new Date(Date.now() - 24 * 60 * 60 * 1000).toJSON(),
+      CandleSince: subDays(Date.now(), 1).toJSON(),
     }),
     []
   );
+  const update = useCallback((delta) => {
+    if (statusesRef.current[delta.market.id] !== delta.market.state) {
+      statusesRef.current = {
+        ...statusesRef.current,
+        [delta.market.id]: delta.market.state,
+      };
+      return false;
+    }
+    return true;
+  }, []);
```

Comment:
suggestion (non-blocking): Is it worth passing an array of dependencies here?

Reply:
yeah, I added missing dependency to the useCallback hook, however hook works exactly the same way like before. It's because `refs` aren't necessary in dependencies. We also don't need here a custom hook, because `update` function isn't reusable